File: sample/src/main/java/com/github/lany192/box/sample/http/ApiService.java
package com.github.lany192.box.sample.http;

import com.github.lany192.box.sample.bean.Area;
import com.github.lany192.box.sample.bean.Result;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;

public interface ApiService {
    /**
     * 获取省市县数据
     */
    @GET("json/city.json")
    Observable<Result<List<Area>>> getCityInfo();
}

File: sample/src/main/java/com/github/lany192/box/sample/http/ApiService.kt
package com.github.lany192.box.sample.http

import com.github.lany192.box.sample.bean.Area
import com.github.lany192.box.sample.bean.Result
import retrofit2.http.GET

interface ApiService {
    /**
     * 获取省市县数据
     */
    @GET("https://xzwcn.oss-cn-shanghai.aliyuncs.com/config/city.json")
    suspend fun cityInfo(): Result<List<Area>>
}